src/@types/index.d.ts

```diff
@@ -40,22 +55,32 @@ interface Server {
 
 // Config/tasks
 // ---------------------------------------------------------------------------
+interface TaskClassification {
+  country_code?: CountryCode[];
+  asn?: ASN[];
+  connection_type?: ConnectionType[];
+  device_type?: DeviceType[];
+  [key: string]: string[] | number[] | undefined;
```

Comment:
Not crazy about (1) the design of allowing arbitrary key-value pairs at the same level as canonical (especially optional) values such as `country_code`. For example, one could specify `countryCode` as an option and the type-checker would not catch it. And (2), since these are all arrays, I would expect these keys to be plural (e.g. `country_codes`, `connection_types`, etc.) Wouldn't it be better to opt for something like:
```
interface AdditionalOptions {
  [key: string: string[] | number[] | undefined;
}
interface TaskClassification {
  country_codes?: CountryCode[];
  asns?: ASN[];
  connection_types?: ConnectionType[];
  device_types?: DeviceType[];
  additional_options: AdditionalOptions;
}
```

Reply:
For 1) this was actually to keep the compiler happy, as it won't let me index the object by string without it i.e. to be able to use the syntax `classification[key]`. Which is needed for the reducer function to work correctly whilst iterating over the keys.

For 2) agreed, however these are the keys returned by the API, therefore would like to maintain parity/consistency.